pytube/__main__.py

```diff
@@ -177,80 +239,60 @@ def descramble(self) -> None:
+                )
+                self._fmt_streams.append(video)
 
-        # load the player_response object (contains subtitle information)
-        if isinstance(self.player_config_args["player_response"], str):
-            self.player_response = json.loads(
-                self.player_config_args["player_response"]
-            )
-        else:
-            self.player_response = self.player_config_args["player_response"]
-        del self.player_config_args["player_response"]
         self.stream_monostate.title = self.title
         self.stream_monostate.duration = self.length
 
-    def prefetch(self) -> None:
-        """Eagerly download all necessary data.
+        return self._fmt_streams
 
-        Eagerly executes all necessary network requests so all other
-        operations don't does need to make calls outside of the interpreter
-        which blocks for long periods of time.
+    def check_availability(self):
+        """Check whether the video is available.
 
-        :rtype: None
+        Raises different exceptions based on why the video is unavailable,
+        otherwise does nothing.
         """
-        self.watch_html = request.get(url=self.watch_url)
-        self.check_availability()
-        self.age_restricted = extract.is_age_restricted(self.watch_html)
-
-        if self.age_restricted:
-            if not self.embed_html:
-                self.embed_html = request.get(url=self.embed_url)
-            self.vid_info_url = extract.video_info_url_age_restricted(
-                self.video_id, self.watch_url
-            )
-            self.js_url = extract.js_url(self.embed_html)
-        else:
-            self.vid_info_url = extract.video_info_url(
-                video_id=self.video_id, watch_url=self.watch_url
-            )
-            self.js_url = extract.js_url(self.watch_html)
-
-        self.initial_data = extract.initial_data(self.watch_html)
-
-        self.vid_info_raw = request.get(self.vid_info_url)
-
-        # If the js_url doesn't match the cached url, fetch the new js and update
-        #  the cache; otherwise, load the cache.
-        if pytube.__js_url__ != self.js_url:
-            self.js = request.get(self.js_url)
-            pytube.__js__ = self.js
-            pytube.__js_url__ = self.js_url
-        else:
-            self.js = pytube.__js__
-
-    def initialize_stream_objects(self, fmt: str) -> None:
-        """Convert manifest data to instances of :class:`Stream <Stream>`.
-
-        Take the unscrambled stream data and uses it to initialize
-        instances of :class:`Stream <Stream>` for each media stream.
+        status, messages = extract.playability_status(self.watch_html)
 
-        :param str fmt:
-            Key in stream manifest (``ytplayer_config``) containing progressive
-            download or adaptive streams (e.g.: ``url_encoded_fmt_stream_map``
-            or ``adaptive_fmts``).
+        for reason in messages:
+            if status == 'UNPLAYABLE':
```

Comment:
can we enum values /w a typeclass or similar :) 
general  string constants scare me

Reply:
the status and reason since they both seem like they can be mostly fixed? 
no clue

Reply:
The status and reason strings are extracted directly from youtube, and are not constants that I randomly decided on